R/germline.R

```diff
@@ -93,19 +76,58 @@ take_first_allele <- function(string) {
   unlist(strsplit(string, ","))[1]
 }
 
-generate_germline_sequence <- function(v_seq, j_seq) {
-  if (is.na(v_seq) || is.na(j_seq)) {
+generate_germline_sequence <- function(seq, v_ref, v_end, cdr3_start, cdr3_end, j_start) {
+  if (is.na(seq) || is.na(v_ref) ||
+    is.na(v_end) || is.na(cdr3_start) || is.na(cdr3_end) || is.na(j_start)) {
     return(NA)
   } else {
-    return(paste(v_seq, j_seq, sep = "..."))
+    cdr3_start %<>% as.numeric()
+    cdr3_end %<>% as.numeric()
+
+    if (v_end <= cdr3_start) {
+      v_part <- v_ref
+    } else {
+      # trim intersection of V and CDR3 from reference V gene
+      v_part <- stringr::str_sub(
+        v_ref, 1,
+        max(0, stringr::str_length(v_ref) - (v_end - cdr3_start))
+      )
+    }
+
+    cdr3_part <- paste(rep("n", cdr3_end - cdr3_start), collapse = "")
+
+    j_part <- stringr::str_sub(seq, max(cdr3_end, j_start))
+
+    paste0(v_part, cdr3_part, j_part) %>%
+      toupper() %>%
+      return()
```

Comment:
Please edit the code according to the style guide
https://style.tidyverse.org/functions.html#return

Reply:
Lack of `return()` looks normal in simple functions, but it's really bad in complex branching, when code modifications (like switching positions of if and else blocks) can lead to bugs